agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4446,10 +4448,10 @@ static uint8_t get_channel_preference(const beerocks::message::sWifiChannel chan
     auto bw                = static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth);
     auto operating_class   = wireless_utils::get_operating_class_by_channel(channel);
 
-    // consider channels in different operating classes as restricted
-    if (operating_class != preference.oper_class) {
-        return preference.preference;
```

Comment:
This should have stayed at INOPERABLE instead of preference.preference. But asserting is fine as well.